src/Transports/MassTransit.ActiveMqTransport/Testing/ActiveMqTestHarness.cs

```diff
@@ -205,9 +212,19 @@ void CleanUpVirtualHost()
 
         static void CleanUpQueue(ISession model, string queueName)
         {
-            model.DeleteDestination(SessionUtil.GetQueue(model, queueName));
-            model.DeleteDestination(SessionUtil.GetQueue(model, $"{queueName}_skipped"));
-            model.DeleteDestination(SessionUtil.GetQueue(model, $"{queueName}_error"));
+            DeleteDestinationIfPossible(model, queueName);
+            DeleteDestinationIfPossible(model, $"{queueName}_skipped");
+            DeleteDestinationIfPossible(model, $"{queueName}_error");
+        }
+
+        private static void DeleteDestinationIfPossible(ISession session, string destinationName)
+        {
+            var destination = SessionUtil.GetQueue(session, destinationName);
+            // AMQP does not support deleting not temporary destinations
+            if (session is not Apache.NMS.AMQP.NmsSession || destination.IsTemporary)
```

Comment:
Then this should use OpenWire to ensure the destination is always deleted, to avoid breaking tests that expect a clean broker.